include/alpaka/vec/Vec.hpp

```diff
@@ -446,141 +407,91 @@ namespace alpaka
-
     namespace traits
     {
         //! ReverseVec specialization for Vec.
         template<typename TDim, typename TVal>
         struct ReverseVec<Vec<TDim, TVal>>
         {
             ALPAKA_NO_HOST_ACC_WARNING
-            ALPAKA_FN_HOST_ACC static auto reverseVec(Vec<TDim, TVal> const& vec) -> Vec<TDim, TVal>
-            {
-                return createVecFromIndexedFn<TDim, alpaka::detail::CreateReverse>(vec);
-            }
-        };
-
-        //! (Non-)ReverseVec specialization for 1D Vec.
-        template<typename TVal>
-        struct ReverseVec<Vec<DimInt<1u>, TVal>>
-        {
-            ALPAKA_NO_HOST_ACC_WARNING
-            ALPAKA_FN_HOST_ACC static auto reverseVec(Vec<DimInt<1u>, TVal> const& vec) -> Vec<DimInt<1u>, TVal>
-            {
-                return vec;
-            }
-        };
-    } // namespace traits
-
-    namespace detail
-    {
-        //! A function object that returns the value at the index from the back of the vector.
-        template<std::size_t Tidx>
-        struct CreateConcat
-        {
-            ALPAKA_NO_HOST_ACC_WARNING
-            template<typename TDimL, typename TDimR, typename TVal>
-            ALPAKA_FN_HOST_ACC static auto create(Vec<TDimL, TVal> const& vecL, Vec<TDimR, TVal> const& vecR) -> TVal
+            ALPAKA_FN_HOST_ACC static constexpr auto reverseVec(Vec<TDim, TVal> const& vec) -> Vec<TDim, TVal>
             {
-                return Tidx < TDimL::value ? vecL[Tidx] : vecR[Tidx - TDimL::value];
+                if constexpr(TDim::value <= 1)
+                {
+                    return vec;
+                }
+                else
+                {
+                    Vec<TDim, TVal> r;
+                    for(typename TDim::value_type i = 0; i < TDim::value; ++i)
+                        r[i] = vec[TDim::value - 1u - i];
+                    return r;
+                }
+                ALPAKA_UNREACHABLE({});
             }
         };
-    } // namespace detail
 
-    namespace traits
-    {
         //! Concatenation specialization for Vec.
         template<typename TDimL, typename TDimR, typename TVal>
         struct ConcatVec<Vec<TDimL, TVal>, Vec<TDimR, TVal>>
         {
             ALPAKA_NO_HOST_ACC_WARNING
-            ALPAKA_FN_HOST_ACC static auto concatVec(Vec<TDimL, TVal> const& vecL, Vec<TDimR, TVal> const& vecR)
-                -> Vec<DimInt<TDimL::value + TDimR::value>, TVal>
+            ALPAKA_FN_HOST_ACC static constexpr auto concatVec(
+                Vec<TDimL, TVal> const& vecL,
+                Vec<TDimR, TVal> const& vecR) -> Vec<DimInt<TDimL::value + TDimR::value>, TVal>
             {
-                return createVecFromIndexedFn<DimInt<TDimL::value + TDimR::value>, alpaka::detail::CreateConcat>(
-                    vecL,
-                    vecR);
+                Vec<DimInt<TDimL::value + TDimR::value>, TVal> r;
+#if BOOST_COMP_NVCC
+#    pragma push
+#    pragma nv_diag_suppress = unsigned_compare_with_zero
```

Comment:
it at least complains about it in the frontend.

Reply:
mhm, IIRC what I've done elsewhere was to protect the loop with something like `if constexpr (TDimL::value > 0) { ... }`

Reply:
sure, that would work as well. but it should not be necessary. a loop like that:
```c++
for (unsigned int i = 0; i < 0; i++) { .. }
```
is fine and will just not execute its body. that's why I chose to suppress the warning rather than adding the conditional, which might maybe cause another warning.